allennlp/service/servable/models/semantic_role_labeler.py

```diff
@@ -1,34 +1,26 @@
-import os
 from typing import Dict
 
-from allennlp.common import Params
-from allennlp.data import Vocabulary
-from allennlp.data.dataset_readers import SrlReader
+from allennlp.data import Vocabulary, Tokenizer, TokenIndexer
 from allennlp.data.fields import TextField, IndexField
-from allennlp.data.tokenizers import Tokenizer, WordTokenizer
-from allennlp.data.token_indexers import TokenIndexer
-from allennlp.models import SemanticRoleLabeler
-from allennlp.service.servable import Servable, JSONDict
+from allennlp.models import Model
+from allennlp.service.servable import Servable, JsonDict
 
 import spacy
 
+@Servable.register("srl")
 class SemanticRoleLabelerServable(Servable):
-    def __init__(self,
-                 tokenizer: Tokenizer,
-                 token_indexers: Dict[str, TokenIndexer],
-                 model: SemanticRoleLabeler) -> None:
-        self.tokenizer = tokenizer
-        self.token_indexers = token_indexers
-        self.model = model
-        self.model.eval()
+    def __init__(self, model: Model, vocab: Vocabulary,
+                 tokenizer: Tokenizer, token_indexers: Dict[str, TokenIndexer]) -> None:
+        super().__init__(model, vocab, tokenizer, token_indexers)
+
         self.nlp = spacy.load('en', parser=False, vectors=False, entity=False)
 
-    def predict_json(self, inputs: JSONDict) -> JSONDict:
+    def predict_json(self, inputs: JsonDict) -> JsonDict:
         sentence = inputs["sentence"]
         tokens = self.tokenizer.tokenize(sentence)
         text = TextField(tokens, token_indexers=self.token_indexers)
 
-        results = {"verbs": []}  # type: JSONDict
+        results = {"verbs": []}  # type: JsonDict
         spacy_doc = self.nlp(sentence)
         for i, word in enumerate(spacy_doc):
```

Comment:
I know this is WIP - moving the default instantiation of the tokenizer / token indexers made me look at this and wonder if there was a better way.  I don't have a better option than what you did, so it seems fine.  But, I noticed an issue with the way this works - you're tokenizing the sentence using `self.tokenizer`, but using `self.nlp` to get tokens for finding verbs.  This might not match (in particular, the spacy tokenizer is not the default in `WordTokenizer`).  That means you should probably just not use `self.tokenizer` at all here, and only rely on `self.nlp` to do the tokenization.  Not sure if that changes any decisions around what should go in the base `DatasetReader` class - probably not.